lib/internal/child_process.js

```diff
@@ -655,6 +655,16 @@ function setupChannel(target, channel) {
     if (message === undefined)
       throw new ERR_MISSING_ARGS('message');
 
+    // non-serializable messages should not reach the remote
+    // end point; as any failure in the strinfication there
+    // will result in error message that is weekly consumable.
+    // So perform a sanity check on message prior to sending.
+    const ret = JSON.stringify(message);
```

Comment:
Calling `JSON.stringify()` twice is wasteful. Cache the result and use it on line 730.

Reply:
unfortunately, the `message` undergoes change in all the possible control flows between here and line 730, so cached value is unusable.

Another alternative is to delay this sanity check to as late as line 730 where there is already a stringification, don't know whether it can have side effects or not - thoughts please?

Reply:
@BridgeAR - I followed your suggestions, PTAL.